CS3219-AY2425S1/cs3219-ay2425s1-project-g50
# Add API gateway

Resolves #14, #19, #20, #24 

## Overview
Previously, our question service was exposed, allowing anyone to use the endpoints. The API gateway provides automatic authentication, as well as provides ingress capabilities.

Currently does not provide load balancing capabilities. Intention is for load balancing to be done by **k8s**.

## Testing instructions
1. Add in `API_GATEWAY_PORT` value to `.env`. Note that whatever port you set here will determine how you access the pages.
2. docker compose up --build.
3. Go to `localhost:$API_GATEWAY_PORT` to access frontend.
4. Play around with backend endpoints through frontend.
5. Play around with backend endpoints through preferred requester, port is always $API_GATEWAY_PORT.
